backend/tracim_backend/command/cleanup.py

```diff
@@ -139,10 +139,13 @@ def take_app_action(self, parsed_args: argparse.Namespace, app_context: AppEnvir
             user_list = []  # type: typing.List[User]
             for login in parsed_args.logins:
                 try:
-                    user = uapi.get_one_by_email(login)
+                    if "@" in login:
```

Comment:
I'm not sure about the way to deal with this, in the specification of the ticket, it was marked to check "@" in login to verificate email and then username. This create a third behavior for a similar things.
- First, this code is currently duplicate at places in command code. It make sense to have it inside userapi.
- Secondly, we do here a check with "@", It's a third way to do similar things in tracim, for api authentication, we do pass both email and username, and for ldap/basic auth authentication, we do multiple check (try except), we should clarify the rule here and refactor in order to have maximum 2 way to do this, with this kind of logic in userapi.

Reply:
And I agree with your remark that we should limit the number of behaviors for login, from what I see there is `UserApi._authenticate` to modify in order to use `get_one_by_login` (and some renaming of email -> login in `_ldap_authenticate`/`_internal_db_authenticate`/`_remote_user_authenticate`). I think this could be done in this PR.

It also would be cleaner to have a login field in the authentication HTTP API instead of email/username which duplicates the behavior to distinguish the username/email (as the frontend must differentiate to give the proper parameters): for this one we should create another issue as the frontend is involved.

BTW, is the remote user authentication method used ? We didn't consider it when adding username support in authentication so it doesn't work with username (`UserApi._remote_authenticate` tries only with `get_one_by_email`).

Reply:
I've done all the refactoring needed and i added support for username in both ldap and remote auth (username is used when "@" is not available in login).

Reply:
just added support for username in ApiToken Auth mecanism too. It make no sense to not be able to login to username-only account with this method.